main.js

```diff
@@ -413,6 +447,27 @@ app.post('/validateEmail', function(req, res, next) {
   })
 });
 
+// DISPLAY LOGIN PAGE
+app.get('/login',function(req,res,next){
+  let context = {};
+  context.title = "Login";
+  res.render('login', context);
+});
+
+// LOGIN Attempt
+app.post('/login', passport.authenticate("local", {
+  successRedirect: "/",
+  failureRedirect: "/login",
+  failureFlash: true
+  })
+);
+
+// LOGOUT
+app.get('/logout', function(req,res){
```

Comment:
Let's use a `DELETE` request instead of `GET` for logout here. See this video for how this is done.

Reply:
Hi hao, can you link the video. I wasn't sure if delete was necessary because I'm using an anchor and not a form.

Reply:
Here you go! https://youtu.be/-RCnNyD0L-s?t=2015

Reply:
It just bothers me that someone could type `http://localhost:3000/logout` and trigger this code path. I think `DELETE` is the right convention to follow, although I am fine with `POST` as well.

Reply:
ok, ill add in the checkauthenticated and checknotauthenticated functions then look into making the anchor into a form to do the delete. Maybe i'll do it on Tuesday. Going to work on my other class. Side note, i was trying to see if i can just import the database connection stuff from main.js but I couldn't figure out how to get it to work. Not sure if i needed to export more then one thing from main.js so i just retyped all the connection stuff to save time.

Reply:
done.